src/frontend/shortlist/src/components/preferences/PreferenceManager.vue

```diff
@@ -0,0 +1,182 @@
+<script>
+import { ref, provide } from 'vue'
+
+export default {
+  name: "PreferenceManager",
+  props: ["title"],
+  emits: ["preferenceSubmitted"],
+  methods: {
+    clicker(event) {
+      
+      if (event == "Next") {
+        this.selectedPage == this.numPages - 1 ? this.selectedPage = this.selectedPage : this.selectedPage++;
+      } else if (event == "Back") {
+        this.selectedPage == 0 ? this.selectedPage = this.selectedPage : this.selectedPage--;
+      } else {
+        this.$emit("preferenceSubmitted");
+      }
+      
+      if (this.selectedPage > 0) {
+        this.actions.back.enabled = true;
+        
+      } else {
+        this.actions.back.enabled = false;
+      }
+
+      if (this.selectedPage == this.numPages - 1) {
+        this.actions.next.enabled = false;
+        this.actions.submit.enabled = true;
+      } else {
+        this.actions.next.enabled = true;
+        this.actions.submit.enabled = false;
+      }
+    }
+  },
+  data() {
+    const actions = {
+      back: {
+        text: "Back",
+        enabled: false,
+        type: "button",
+      },
+      next: {
+        text: "Next",
+        enabled: false,
+        type: "button",
+      },
+      submit: {
+        text: "Submit",
+        enabled: false,
+        type: "submit",
+      }
+    };
+
+    if (this.numPages == 1) {
+      actions.submit.enabled = true;
+    } else {
+      actions.next.enabled = true;
+    }
+
+    return {
+      actions
+    }
+  },
+  setup(props, { slots }, ) {
+    let preferencePages = ref(slots.default());
+
+    let numPages = preferencePages.value.length;
+    let selectedPage = ref(0);
```

Comment:
At some point we should figure out whether the ref and the inject were overkill here.  there might be a simpler vmodel or vbind approach for choosing which page is active/visible